### PR TITLE
MODTLR-68: Allowed service points - only use lending side circulation rules

### PR DESCRIPTION
**Pull request by imerabishvili**
## Purpose
https://folio-org.atlassian.net/browse/MODTLR-68

## Approach
1. Updated existing implementation to remove checks based on routing service points.
2. Used approach to fetch all valid service points in corresponding tenants.

### CODE REVIEW

File: src/main/java/org/folio/service/impl/AllowedServicePointsForItemLevelRequestService.java [hunk 1]
```diff
@@ -41,15 +41,15 @@ protected Collection<String> getLendingTenants(AllowedServicePointsRequest reque
   }
 
   @Override
-  protected AllowedServicePointsResponse getAllowedServicePointsFromLendingTenant(
+  protected AllowedServicePointsResponse getAllowedServicePointsFromTenant(
     AllowedServicePointsRequest request, String patronGroupId, String tenantId) {
 
     log.info("getAllowedServicePointsFromLendingTenant:: parameters: request: {}, " +
```
Comment (alexanderkurash) [inline src/main/java/org/folio/service/impl/AllowedServicePointsForItemLevelRequestService.java:47]: You've changed the name of the method, but still logging the old name

File: src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java [hunk 2]
```diff
@@ -49,107 +52,70 @@ private AllowedServicePointsResponse getForCreate(AllowedServicePointsRequest re
...
-        "ECS TLR for primary request %s was not found", primaryRequestId)));
-
-    log.info("findEcsTlr:: ECS TLR found: {}", ecsTlr.getId());
-    return ecsTlr;
-  }
-
-  private AllowedServicePointsResponse getAllowedServicePointsFromBorrowingTenant(
-    AllowedServicePointsRequest request) {
 
     log.info("getForReplace:: fetching allowed service points from borrowing tenant");
```
Comment (alexanderkurash) [inline src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java:92]: We're not fetching them from the borrowing tenant anymore

File: src/main/java/org/folio/service/impl/AllowedServicePointsForItemLevelRequestService.java (hunk 1 above)
Comment (imerabishvili) [reply src/main/java/org/folio/service/impl/AllowedServicePointsForItemLevelRequestService.java]: fixed

File: src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java (hunk 2 above)
Comment (imerabishvili) [reply src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java]: fixed

File: src/main/java/org/folio/controller/AllowedServicePointsController.java [hunk 3]
```diff
@@ -38,20 +36,14 @@ public ResponseEntity<AllowedServicePointsResponse> getAllowedServicePoints(Stri
...
-  }
-
-  private AllowedServicePointsService getAllowedServicePointsService(
-    AllowedServicePointsRequest request) {
-    return request.isForTitleLevelRequest()
+    var allowedServicePointsService = request.isForTitleLevelRequest()
       ? allowedServicePointsForTitleLevelRequestService
       : allowedServicePointsForItemLevelRequestService;
+    var response = allowedServicePointsService.getAllowedServicePoints(request);
+    return ResponseEntity.status(OK).body(response);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/controller/AllowedServicePointsController.java:46]:
```suggestion
    var response = allowedServicePointsService.getAllowedServicePoints(request);
    
    return ResponseEntity.status(OK).body(response);
```

Comment (roman-barannyk) [inline src/main/java/org/folio/controller/AllowedServicePointsController.java:46]:
```suggestion

    return ResponseEntity.status(OK).body(allowedServicePointsService
      .getAllowedServicePoints(request));
```

File: src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java [hunk 4]
```diff
@@ -49,58 +52,62 @@ private AllowedServicePointsResponse getForCreate(AllowedServicePointsRequest re
...
+      return;
+    }
+    toAdd.stream()
       .filter(Objects::nonNull)
-      .flatMap(Collection::stream)
-      .anyMatch(Objects::nonNull);
-
-    log.info("isAvailableInLendingTenant:: result: {}", availabilityCheckResult);
-    return availabilityCheckResult;
+      .forEach(t -> servicePoints.put(t.getId(), t));
```
Comment (roman-barannyk) [inline src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java:81]:
```suggestion
      .forEach(allowedSp -> servicePoints.put(allowedSp.getId(), allowedSp));
```

**Review by roman-barannyk — APPROVED** [review]

File: src/main/java/org/folio/controller/AllowedServicePointsController.java (hunk 3 above)
Comment (imerabishvili) [reply src/main/java/org/folio/controller/AllowedServicePointsController.java]: done

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**